src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +164,1314 @@ static DamageTable D_80BBF858 = {
+         *	if ((actorType % 2) == 0)
+         * However, the original code checks the negative case whereas the above does not.
+         * So unfortunately, this is the only way to get it to match.
+         */
+        if (!(actorType % 2)) {
+            this->actionFlags |= EN_S_GORO_ACTIONFLAG_SNOREPHASE;
+        }
+        this->actionFlags |= EN_S_GORO_ACTIONFLAG_SUPPRESS_SNORE;
+        this->actionFunc = EnSGoro_Sleep;
+        return true;
+    }
+    return false;
+}
+
+void EnSGoro_UpdateToHandtapAnimation(EnSGoro* this) {
+    s16 curFrame = this->skelAnime.curFrame;
+    s16 lastFrame = Animation_GetLastFrame(sAnimationInfo[this->animInfoIndex].animation);
+
+    if ((this->animInfoIndex != EN_S_GORO_ANIM_STAND_HANDTAP) && (curFrame == lastFrame)) {
+        this->animInfoIndex = EN_S_GORO_ANIM_STAND_HANDTAP;
+        SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+    }
+}
+
+void EnSGoro_UpdateSleepyAnimation(EnSGoro* this) {
+    s16 curFrame = this->skelAnime.curFrame;
+    s16 lastFrame = Animation_GetLastFrame(sAnimationInfo[this->animInfoIndex].animation);
+
+    if (this->animInfoIndex != EN_S_GORO_ANIM_SLEEPY) {
+        if (curFrame == lastFrame) {
+            this->animInfoIndex = EN_S_GORO_ANIM_SLEEPY;
+            SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+        }
+    } else if (curFrame == lastFrame) {
+        this->actionFlags &= ~EN_S_GORO_ACTIONFLAG_TIRED;
+        this->animInfoIndex = EN_S_GORO_ANIM_IDLE_STAND;
+        SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+    }
+}
+
+void EnSGoro_UpdateToIdleAnimation(EnSGoro* this) {
+    s16 curFrame = this->skelAnime.curFrame;
+    s16 lastFrame = Animation_GetLastFrame(sAnimationInfo[this->animInfoIndex].animation);
+
+    if ((this->animInfoIndex != EN_S_GORO_ANIM_IDLE_STAND) && (curFrame == lastFrame)) {
+        this->animInfoIndex = EN_S_GORO_ANIM_IDLE_STAND;
+        SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+    }
+}
+
+void EnSGoro_UpdateCollider(EnSGoro* this, PlayState* play) {
+    Vec3f world_pos = this->actor.world.pos;
+    f32 radius = 24.0f;
+    f32 height = 62.0f;
+
+    this->collider.dim.pos.x = world_pos.x;
+    this->collider.dim.pos.y = world_pos.y;
+    this->collider.dim.pos.z = world_pos.z;
+    this->collider.dim.radius = radius;
+    this->collider.dim.height = height;
+
+    //! @bug: It is not clear what this is for.
+    if ((s32)this != -0x190) {
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBCA80.s")
+void EnSGoro_UpdateEyes(EnSGoro* this) {
+    if (this->actionFlags & EN_S_GORO_ACTIONFLAG_EYESOPEN) {
+        if (DECR(this->eyeTimer) == 0) {
+            this->eyeTexIndex++;
+            if (this->eyeTexIndex >= EN_S_GORO_EYETEX_INDEX_MAX) {
```

Comment:
```suggestion
            if (this->eyeTexIndex > EN_S_GORO_EYETEX_CLOSED) {
```
Personally I think I like this better than the separate max define.